client/src/lib/votecounting.ts

```diff
@@ -77,19 +74,59 @@ export function getVotesByPrecinct({
   return votesByPrecinct
 }
 
+export const getVotesByScanner: VotesByFunction = ({
+  election,
+  castVoteRecords,
+}) => {
+  const votesByScanner: VotesByFilter = {}
+  castVoteRecords.forEach((CVR) => {
```

Comment:
It might be easier (and more extensible and easier to debug / test) to have a single tabulation function that takes a categorization function as an input, and that function is responsible for bucketizing the CVRs. For example:

```
getVotesByCategorization = ({election, cvrs, categorizationFunction}) => {
     ...
     const category = categorizationFunction(cvr)
     ...
}
```

and for example:

```
const categorizeByScannerId = (cvr) => {return cvr._scannerId}
```

Reply:
Totally agree. I was trying to channel the quick-Ben-way of updating this code. 😉  I was tempted to refactor, but figured that could be another pass.